compiler/ml/typedtree.mli

```diff
@@ -174,8 +174,8 @@ and expression_desc =
 
             [Texp_match (E0, [(P1, E1); (P2, E2)], [(P3, E3)], _)]
          *)
-  | Texp_try of expression * case list
-      (** try E with P1 -> E1 | ... | PN -> EN *)
+  | Texp_try of expression * case list * expression option
```

Comment:
nit: get assistant to use a record in a hypothetical follow-up PR, if this goes ahead